src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.Abstractions;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.Messaging.Pumps.ServiceBus
+{
+    public abstract class AzureServiceBusMessagePump<TMessage> : MessagePump<TMessage, AzureServiceBusMessageContext>
+    {
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="configuration">Configuration of the application</param>
+        /// <param name="logger">Logger to write telemetry to</param>
+        protected AzureServiceBusMessagePump(IConfiguration configuration, ILogger logger)
+            : base(configuration, logger)
+        {
+        }
+
+        /// <summary>
+        ///     Path of the entity to process
+        /// </summary>
+        public string EntityPath { get; private set; }
+
+        /// <summary>
+        ///     Service Bus namespace that contains the entity
+        /// </summary>
+        public string Namespace { get; private set; }
+
+        /// <inheritdoc/>
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            Logger.LogInformation("Creating message pump");
+            var messageReceiver = CreateMessageReceiver();
+            Logger.LogInformation("Starting message pump on entity path {EntityPath} in namespace {Namespace}", EntityPath, Namespace);
+
+            // TODO: Message pump options to not delete for example
+            var messageHandlerOptions = new MessageHandlerOptions(HandleReceivedException);
```

Comment:
what's the policy on calling base methods? I'd prefer the use of base.
eg: base.HandleReceivedException

Reply:
No explicit policy yet, right @stijnmoreels?

We can change it since it's more explicit, but also tricky since if we override it, we have to check all usages of it.

I'm happy with either approach.

Reply:
> No explicit policy yet, right @stijnmoreels?

No, I think not, though I may lean to @wseye 's approach but only if indeed overriding is off the table.

Reply:
Advocate of the devil - What if somebody wants to extend AzureServiceBusMessagePump and overrides it? AFAIK it will not be taken into account.

Reply:
That is true, but then he isn't calling a method of the base method but his own overridden method. But I understand that it could introduce an error. 
I'm happy either way, just asking if there was a policy :)

Reply:
I'd suggest to stick with this then, for now.

Ok for you guys?

Reply:
Let's turn the question around, what is the benefit of specifying other than being explicit?

Reply:
Any feedback?

Reply:
I don't see any benefit.

Reply:
Resolving this one since there is no real benefit